app/js/services/addon.js

```diff
@@ -140,3 +140,58 @@ AddonService.uninstall = function(origin) {
     }).catch(reject);
   });
 };
+
+AddonService.getMerger = function() {
+  return new Promise((resolve, reject) => {
+    var name = window.prompt('Enter a name for this add-on', 'Merged Addon ' + new Date().toISOString());
+    if (!name) {
+      reject();
+      return;
+    }
+
+    var merger = new AddonMerger(name);
+    merger.manifest.customizations = [{
+      filter: window.location.origin,
+      scripts: ['main.js']
+    }];
+
+    resolve(merger);
+  });
+};
+
+AddonService.merge = function(origins) {
+  return new Promise((resolve, reject) => {
+    AddonService.getMerger().then((merger) => {
+      var blobs = [];
+      var addons = [];
+
+      origins.forEach((origin) => {
+        AddonService.getAddon(origin).then((addon) => {
+          addons.push(addon);
+
+          addon.export().then((blob) => {
+            blobs.push(blob);
+
+            if (blobs.length === origins.length) {
+              blobs.forEach(blob => merger.add(blob));
+
+              merger.merge((arrayBuffer) => {
+                if (!arrayBuffer) {
+                  reject();
+                  return;
+                }
+
+                addons.forEach((addon) => {
+                  navigator.mozApps.mgmt.uninstall(addon);
```

Comment:
This can fail. We should handle it more explicitly. Perhaps we could re-prompt them or something until they accept it, or abort the whole thing if they reject any.